sdk/schemaregistry/azure-data-schemaregistry-jsonschema/src/test/java/com/azure/data/schemaregistry/jsonschema/SchemaRegistryJsonSchemaSerializerTests.java

```diff
@@ -0,0 +1,372 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests {@link SchemaRegistryJsonSchemaSerializer}.
+ */
+public class SchemaRegistryJsonSchemaSerializerTests {
+    private static final String ADDRESS_SCHEMA_ID = "address-schema-id";
+    private static final String SCHEMA_GROUP = "test-group";
+    private static final String CONTENT_TYPE_WITH_ID = ContentType.APPLICATION_JSON + "+" + ADDRESS_SCHEMA_ID;
+    private static final SerializerAdapter SERIALIZER_ADAPTER = JacksonAdapter.createDefaultSerializerAdapter();
+
+    @Mock
+    private SchemaRegistryAsyncClient registryAsyncClient;
+    @Mock
+    private JsonSchemaGenerator jsonSchemaGenerator;
+    @Mock
+    private SchemaProperties schemaProperties;
+
+    private AutoCloseable mocksCloseable;
+
+    private final Address address = new Address();
+
+    @BeforeEach
+    public void beforeEach() {
+        this.mocksCloseable = MockitoAnnotations.openMocks(this);
+
+        address.setStreetName("Rodeo");
+        address.setNumber(20);
+        address.setStreetType("Street");
+
+        when(schemaProperties.getId()).thenReturn(ADDRESS_SCHEMA_ID);
+        when(schemaProperties.getFormat()).thenReturn(SchemaFormat.JSON);
+        when(schemaProperties.getGroupName()).thenReturn(SCHEMA_GROUP);
+        when(schemaProperties.getVersion()).thenReturn(1);
+    }
+
+    @AfterEach
+    public void afterEach() throws Exception {
+        if (mocksCloseable != null) {
+            mocksCloseable.close();
+        }
+
+        Mockito.framework().clearInlineMock(this);
+    }
+
+    /**
+     * Tests that constructor throws exceptions we expect.
+     */
+    @Test
+    public void nullConstructorArgs() {
+        // Arrange
+        SerializerOptions options = new SerializerOptions("foo", false, 10, SERIALIZER_ADAPTER);
+
+        // Act & Assert
+        assertThrows(NullPointerException.class,
+            () -> new SchemaRegistryJsonSchemaSerializer(null, jsonSchemaGenerator, options));
+        assertThrows(NullPointerException.class,
+            () -> new SchemaRegistryJsonSchemaSerializer(registryAsyncClient, null, options));
+        assertThrows(NullPointerException.class,
+            () -> new SchemaRegistryJsonSchemaSerializer(registryAsyncClient, jsonSchemaGenerator, null));
+    }
+
+    /**
+     * Serializes an object, and registers the schema when auto-register is set to true.
+     */
+    @Test
+    public void serializeNoArgConstructor() throws IOException {
+        // Arrange
+        SerializerOptions serializerOptions = new SerializerOptions(SCHEMA_GROUP, true,
+            100, SERIALIZER_ADAPTER);
+
+        when(registryAsyncClient.registerSchema(eq(SCHEMA_GROUP), eq(Address.class.getName()), any(),
+            eq(SchemaFormat.JSON))).thenAnswer(invocation -> Mono.just(schemaProperties));
+
+        SchemaRegistryJsonSchemaSerializer serializer = new SchemaRegistryJsonSchemaSerializer(registryAsyncClient,
```

Comment:
Since the ctor is package private, can we use the builder to create the serializer to also test the builder?

Reply:
I can add another class to test the builder. I want to scope my testing to the class under test.